--- a/runtime/BUILD.bazel	2023-11-24 16:35:19
+++ b/runtime/BUILD.bazel	2023-11-24 16:36:24
@@ -26,7 +26,7 @@
     deps = [
         "//internal/httprule",
         "//utilities",
-        "@go_googleapis//google/api:httpbody_go_proto",
+        "@@org_golang_google_genproto_googleapis_api//httpbody",
         "@org_golang_google_grpc//codes",
         "@org_golang_google_grpc//grpclog",
         "@org_golang_google_grpc//health/grpc_health_v1",
