diff -urN a/raftpb/BUILD.bazel b/raftpb/BUILD.bazel
--- a/raftpb/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/raftpb/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -1,4 +1,5 @@
 load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+load("@rules_proto//proto:defs.bzl", "proto_library")
 
 go_library(
     name = "raftpb",
@@ -28,3 +29,12 @@
     srcs = ["confstate_test.go"],
     embed = [":raftpb"],
 )
+
+# keep
+proto_library(
+    name = "raftpb_proto",
+    srcs = ["raft.proto"],
+    deps = ["@com_github_gogo_protobuf//gogoproto:gogo_proto"],
+    import_prefix = "etcd/raft/v3/",
+    visibility = ["//visibility:public"],
+)
