diff --git a/src/pd/client.rs b/src/pd/client.rs
index 06240ad..206737c 100644
--- a/src/pd/client.rs
+++ b/src/pd/client.rs
@@ -11,6 +11,7 @@ use futures::stream::BoxStream;
 use log::info;
 use tokio::sync::{watch, Mutex, RwLock};
 use tonic::codegen::http::uri::PathAndQuery;
+use tonic_prost::ProstCodec;
 
 use crate::compat::stream_fn;
 use crate::kv::{ReplicaReadConfig, ReplicaReadType};
@@ -566,7 +567,7 @@ async fn send_store_health_check(
                 service: String::new(),
             }),
             PathAndQuery::from_static("/grpc.health.v1.Health/Check"),
-            tonic::codec::ProstCodec::<HealthCheckRequest, HealthCheckResponse>::default(),
+            ProstCodec::<HealthCheckRequest, HealthCheckResponse>::default(),
         )
         .await
         .map_err(crate::Error::from)
diff --git a/src/store/mockserver.rs b/src/store/mockserver.rs
index 6986fbc..e4092fc 100644
--- a/src/store/mockserver.rs
+++ b/src/store/mockserver.rs
@@ -21,6 +21,7 @@ use tokio_stream::StreamExt;
 use tonic::codegen::{http, Body, BoxFuture, Service, StdError};
 use tonic::transport::server::{Connected, TcpConnectInfo};
 use tonic::transport::Server;
+use tonic_prost::ProstCodec;
 
 use crate::proto::{coprocessor, kvrpcpb, tikvpb};
 
@@ -300,7 +301,7 @@ where
     B: Body + Send + 'static,
     B::Error: Into<StdError> + Send + 'static,
 {
-    type Response = http::Response<tonic::body::BoxBody>;
+    type Response = http::Response<tonic::body::Body>;
     type Error = Infallible;
     type Future = BoxFuture<Self::Response, Self::Error>;
 
@@ -315,11 +316,9 @@ where
                     core: self.core.clone(),
                 };
                 Box::pin(async move {
-                    Ok(
-                        tonic::server::Grpc::new(tonic::codec::ProstCodec::default())
-                            .unary(service, request)
-                            .await,
-                    )
+                    Ok(tonic::server::Grpc::new(ProstCodec::default())
+                        .unary(service, request)
+                        .await)
                 })
             }
             "/tikvpb.Tikv/KvPrewrite" => {
@@ -327,11 +326,9 @@ where
                     core: self.core.clone(),
                 };
                 Box::pin(async move {
-                    Ok(
-                        tonic::server::Grpc::new(tonic::codec::ProstCodec::default())
-                            .unary(service, request)
-                            .await,
-                    )
+                    Ok(tonic::server::Grpc::new(ProstCodec::default())
+                        .unary(service, request)
+                        .await)
                 })
             }
             "/tikvpb.Tikv/CoprocessorStream" => {
@@ -339,11 +336,9 @@ where
                     core: self.core.clone(),
                 };
                 Box::pin(async move {
-                    Ok(
-                        tonic::server::Grpc::new(tonic::codec::ProstCodec::default())
-                            .server_streaming(service, request)
-                            .await,
-                    )
+                    Ok(tonic::server::Grpc::new(ProstCodec::default())
+                        .server_streaming(service, request)
+                        .await)
                 })
             }
             "/tikvpb.Tikv/BatchCommands" => {
@@ -351,11 +346,9 @@ where
                     core: self.core.clone(),
                 };
                 Box::pin(async move {
-                    Ok(
-                        tonic::server::Grpc::new(tonic::codec::ProstCodec::default())
-                            .streaming(service, request)
-                            .await,
-                    )
+                    Ok(tonic::server::Grpc::new(ProstCodec::default())
+                        .streaming(service, request)
+                        .await)
                 })
             }
             _ => Box::pin(async move {
@@ -363,7 +356,7 @@ where
                     .status(200)
                     .header("grpc-status", "12")
                     .header("content-type", "application/grpc")
-                    .body(tonic::body::empty_body())
+                    .body(tonic::body::Body::empty())
                     .unwrap())
             }),
         }
