diff --git a/src/redact.rs b/src/redact.rs
index d6d80dc..a2ff659 100644
--- a/src/redact.rs
+++ b/src/redact.rs
@@ -83,6 +83,9 @@ pub fn redact_key_error_if_necessary(error: &mut ProtoKeyError) {
     if let Some(assertion_failed) = &mut error.assertion_failed {
         redact_nonempty(&mut assertion_failed.key);
     }
+    if let Some(shared_lock_lost) = &mut error.shared_lock_lost {
+        redact_nonempty(&mut shared_lock_lost.key);
+    }
     if let Some(lock_info) = error
         .primary_mismatch
         .as_mut()
@@ -191,6 +194,10 @@ mod tests {
                 key: b"assertion".to_vec(),
                 ..Default::default()
             }),
+            shared_lock_lost: Some(kvrpcpb::SharedLockLost {
+                key: b"shared-lock-lost".to_vec(),
+                start_ts: 7,
+            }),
             primary_mismatch: Some(kvrpcpb::PrimaryMismatch {
                 lock_info: Some(lock()),
             }),
@@ -213,6 +220,7 @@ mod tests {
         assert_eq!(error.commit_ts_expired.as_ref().unwrap().key, b"?");
         assert_eq!(error.txn_not_found.as_ref().unwrap().primary_key, b"?");
         assert_eq!(error.assertion_failed.as_ref().unwrap().key, b"?");
+        assert_eq!(error.shared_lock_lost.as_ref().unwrap().key, b"?");
         assert_lock(
             error
                 .primary_mismatch
